### ISSUE TITLE
Error messages when using type tagging are confusing

### ISSUE THREAD

**Issue by coreywoodfield** [yakivy/jam#6]
```
import com.softwaremill.tagging._

trait Test
class A(b: String @@ Test)
trait Module {
  val string: String = "".taggedWith[Test] // type here should be string: String @@ Test
  val a: A = jam.brew[A]
}
```
Expected: Something like `No injection candidates found for (A).b`
Actual:
```
[Error] type mismatch;
[Error]  found   : Module.this.string.type (with underlying type String)
[Error]  required: String @@ Test
[Error]     (which expands to)  String with com.softwaremill.tagging.Tag[Test]
```
It seems like the the macro finds and uses `string` (even though it's not the right type) and then the compiler itself throws an error. I would expect the macro to not use `string` because it's not typed correctly

**Comment by srollins-lucid** [comment]
I suspect this is related to #5, if I had to guess? It's probably grabbing the `String` from `toString()`.

Also, might I propose that the _actual_ expected outcome here is that this is valid and injects the variable `string`, though even while the macro can't handle tags it should still at least give a more intuitive error.

**Comment by coreywoodfield** [comment]
Nope, you can see that it's grabbing `Module.this.string.type`

**Comment by yakivy** [comment]
Should be fixed in `0.0.8`
